src/OracleService/Protocols/OracleNeoFSProtocol.cs

```diff
@@ -76,36 +86,37 @@ private Task<byte[]> GetAsync(Uri uri, string host, CancellationToken cancellati
             };
         }
 
-        private static async Task<byte[]> GetPayloadAsync(Client client, Address addr, CancellationToken cancellation)
+        private static async Task<string> GetPayloadAsync(Client client, Address addr, CancellationToken cancellation)
         {
             Object obj = await client.GetObject(cancellation, new GetObjectParams() { Address = addr }, new CallOptions { Ttl = 2 });
-            return obj.Payload.ToByteArray();
+            return obj.Payload.ToString(Utility.StrictUTF8);
         }
 
-        private static Task<byte[]> GetRangeAsync(Client client, Address addr, string[] ps, CancellationToken cancellation)
+        private static async Task<string> GetRangeAsync(Client client, Address addr, string[] ps, CancellationToken cancellation)
         {
             if (ps.Length == 0) throw new FormatException("missing object range (expected 'Offset|Length')");
             Range range = ParseRange(ps[0]);
-            return client.GetObjectPayloadRangeData(cancellation, new RangeDataParams() { Address = addr, Range = range }, new CallOptions { Ttl = 2 });
+            var res = await client.GetObjectPayloadRangeData(cancellation, new RangeDataParams() { Address = addr, Range = range }, new CallOptions { Ttl = 2 });
+            return Utility.StrictUTF8.GetString(res);
         }
 
-        private static async Task<byte[]> GetHeaderAsync(Client client, Address addr, CancellationToken cancellation)
+        private static async Task<string> GetHeaderAsync(Client client, Address addr, CancellationToken cancellation)
         {
             var obj = await client.GetObjectHeader(cancellation, new ObjectHeaderParams() { Address = addr }, new CallOptions { Ttl = 2 });
-            return Utility.StrictUTF8.GetBytes(obj.ToString());
+            return JObject.Parse(obj.ToString()).ToString();
```

Comment:
Can we read the original byte array directly? Otherwise it is converted many times.

Reply:
Do you mean `obj.ToByteArray()`? It's a byte array in protobuf encoding which we can't handle in contract.

Reply:
There is no way to read the original bytes?

Reply:
Fixed. `ToJson` added. @ProDog Can you test again?

Reply:
Tested OK.